foundation/foundation-drift-detection/aissemble-foundation-drift-detection-client-python/src/drift/detection/rest/client/drift_rest_client.py

```diff
@@ -67,11 +67,9 @@ def invoke_drift(self, policy_identifier, input, control):
         drift_data_input = DriftDataInput(input=input, control=control)
         query_params = {"policyIdentifier": policy_identifier}
         data_headers = {"Content-type": "application/json"}
-        # uncomment below for debugging purposes
-        # print('JSON being sent to drift invocation service: ' + drift_data_input.json())
         return requests.post(
             hostname + "/invoke-drift",
             params=query_params,
-            data=drift_data_input.json(),
+            data=drift_data_input.model_dump_json(serialize_as_any=True),
```

Comment:
@ewilkins-csi Hi! I have been investigating on specifying schema but seems like might not be doable given that one of class have recursive call of object. I also tried consolidating all three classes into one and use model_serializer but no luck there too =(
Looks like they made it to default to prevent any security risk of exposing any PIIs or sensitive information like password. But given that our subclasses doesn't seem to contain any of that, I would say we can keep duck typing. see [here](https://docs.pydantic.dev/latest/concepts/serialization/#serializing-with-duck-typing)
If not, I am open for any implementation ideas!

Reply:
Ok. Thanks for giving it a shot! Since this is just a client and not intended to really be extended, I think it's fair to say our usage is generally safe from the security concern.